browser/ui/BUILD.gn

```diff
@@ -255,6 +255,12 @@ source_set("ui") {
     ]
   }
 
+  if (enable_extensions) {
```

Comment:
I think ntp sponsored images feature is not related with extensions?

Reply:
If we don't have extensions we can't load crx components, right?

Reply:
Well, `ntp_sponsored_images` module doesn't use extension related thing.
I think `component` and `component extension` are different and component just use crx file format.

Reply:
Also curious if this flag is correct - I think it's often used for Android Core to block out extension related features. If we want to use the same mechanism there, I don't think we need this guard.

For example, notice the guards around extension specific items here- but not for other components, like adblock
https://github.com/brave/brave-core/blob/414f559853321386d1b140c7edac1ef0cdfef4f5/browser/brave_browser_process_impl.cc#L162-L181

cc: @bbondy @bridiver

Reply:
Yes this is right, we should remove it. Android supports extensions and we will aim to use all or most of the c++ from this PR for android when it's ported to brave-core. Probably not the http image source if the android frontend will use native views.